packages/pwa-buildpack/src/WebpackTools/PWADevServer.js

```diff
@@ -75,84 +73,97 @@ const PWADevServer = {
             );
         });
     },
-    async findFreeHostname(identifier, times = 0) {
-        const maybeHostname =
-            identifier + (times ? times : '') + '.local.pwadev';
-        // if it has a port, it exists
-        const exists = await PWADevServer.portsByHostname.get(maybeHostname);
-        if (!exists) {
-            debug(
-                `findFreeHostname: ${maybeHostname} unbound to port and available`
-            );
-            return maybeHostname;
+    getUniqueSubdomain(customName) {
+        let name = PWADevServer.DEFAULT_NAME;
+        if (typeof customName === 'string') {
+            name = customName;
         } else {
-            debug(`findFreeHostname: ${maybeHostname} bound to port`, exists);
-            if (times > 9) {
-                throw Error(
+            const pkgLoc = join(process.cwd(), 'package.json');
+            try {
+                // eslint-disable-next-line node/no-missing-require
+                const pkg = require(pkgLoc);
+                if (!pkg.name) {
+                    throw new Error(
+                        `package.json has ${pkg.name} "name" field!`
+                    );
+                }
+                name = pkg.name;
+            } catch (e) {
+                console.warn(
                     debug.errorMsg(
-                        `findFreeHostname: Unable to find a free hostname after 9 tries. You may want to delete your database file at ${GlobalConfig.getDbFilePath()} to clear out old developer hostname entries. (Soon we will make this easier and more automatic.)`
-                    )
+                        `createUniqueSubdomain(): Using default "${name}" prefix. Could not autodetect theme name from package.json`
```

Comment:
The name of this method appears to be `getUniqueSubdomain`, not `createUniqueSubdomain`.

Reply:
Perils of using static refactoring tools. Thanks!